src/medal_widget/events.cljs

```diff
@@ -9,3 +9,18 @@
  ::initialize-db
  (fn-traced [_ _]
    db/default-db))
+
+(re-frame/reg-event-db
+ ::sort-it
+ (fn [db [_ item]]
+   (let [current-data (:medals-data db)
+         sorted (if (= item "gold")
+                  (sort #(or (> (get %1 (keyword item)) (get %2 (keyword item)))
+                             (and (= (get %1 (keyword item)) (get %2 (keyword item)))
+                                  (> (get %1 :total) (get %2 :total)))) current-data)
+                  (sort #(or (> (get %1 (keyword item)) (get %2 (keyword item)))
+                             (and (= (get %1 (keyword item)) (get %2 (keyword item)))
+                                  (> (get %1 :gold) (get %2 :gold)))) current-data))]
```

Comment:
You can do the sort in a much easier way using [sort-by](https://clojuredocs.org/clojure.core/sort-by#example-542692cbc026201cdc326c2c) and [juxt](https://clojuredocs.org/clojure.core/juxt)